dask_expr/_expr.py

```diff
@@ -35,7 +35,10 @@
 no_default = "__no_default__"
 
 
-class Expr:
+from dask.typing import DaskGraph
+# Note: subclassing isn't required. This is just for the prototype to have a
+# check for abstractmethods but the runtime checks for duck-typing/protocol only
+class Expr(DaskGraph):
```

Comment:
as the note says, there is no need for subclassing. this is just for prototyping (or we keep it, no strong preferences)